src/hazelweb/gui/UHCode.re

```diff
@@ -52,11 +52,14 @@ let has_child_clss = (has_child: bool) =>
 
 let caret_from_pos = (x: float, y: float): Vdom.Node.t => {
   let pos_attr =
-    Vdom.Attr.style(
-      Css_gen.combine(
-        Css_gen.left(`Px(int_of_float(x))),
-        Css_gen.top(`Px(int_of_float(y))),
-      ),
+    Vdom.Attr.create(
+      "style",
+      "left: "
```

Comment:
why aren't you using Css_gen here?

Reply:
Css_gen only allows px to be ints https://ocaml.janestreet.com/ocaml-core/latest/doc/virtual_dom/Css_gen/Length/index.html, and that rounding is what is causing the issues you were seeing. 
We could maybe convert to pt measurements, but I am pretty sure you would need the dpi.

Reply:
What happens if you do proper rounding using `int_of_float(Float.round(x))`? `int_of_float` always truncates

Reply:
different browsers seem to implement decimal pixel values slightly differently in terms of how they round those to actual pixels. rounding ourselves may at least make behavior consistent between browsers?

Reply:
can we set the font-size in a way that character width is always an integer?

Reply:
"Setting the font size in pixel values (px) is a good choice when you need pixel accuracy. A px value is static. This is an OS-independent and cross-browser way of literally telling the browsers to render the letters at exactly the number of pixels in height that you specified."

Reply:
The other thing you'd want to tweak is the font size ratio (w.r.t. to the font size) so that both reduced font and surrounding padding can be whole numbers of pixels. If we do this, then let's also fix the issue of potential constant skew of font size ratio between `UHCode.re` and `style.css` (forgot to check for this in original PR).

Reply:
The code is just setting the width of the HoleLabel and centering the number inside instead of trying to figure out padding with rounding numbers.

Reply:
Good idea 👍